contrib/timelapse.lua

```diff
@@ -0,0 +1,463 @@
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+--[[About this plugin
+This plugin will add the new export module "timelapse".
+   
+----REQUIRED SOFTWARE----
+ffmpeg
+
+----USAGE----
+1. Go to Lighttable 
+2. Select images you want to use as a timelapse frames
+3. In image export module select 'timelapse'
+4. Configure you video settings
+5. Export
+
+----WARNING----
+This script has been tested under Linux only
+]]
+
+local dt = require "darktable"
+local df = require "lib/dtutils.file"
+local dsys = require "lib/dtutils.system"
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{5,0,0})
+
+local MODULE_NAME = 'timelapse'
+gettext.bindtextdomain(MODULE_NAME, dt.configuration.config_dir..'/lua/')
+
+local function _(msgid)
+  return gettext.dgettext(MODULE_NAME, msgid)
+end
+
+local OS_PATH_SEPARATOR = dt.configuration.running_os == "windows" and  "\\"  or  "/"
+
+---- DECLARATIONS
+
+local resolutions = {
+  ['QVGA'] = {
+    ['label'] = 'QVGA 320x240 (4:3)',
+    ['w'] = 320,
+    ['h'] = 240
+  },
+  ['HVGA'] = {
+    ['label'] = 'HVGA 480x320 (3:2)',
+    ['w'] = 480,
+    ['h'] = 320
+  },
+  ['VGA'] = {
+    ['label'] = 'VGA 640x480 (4:3)',
+    ['w'] = 640,
+    ['h'] = 480
+  },
+  ['HDTV 720p'] = {
+    ['label'] = 'HDTV 720p 1280x720 (16:9)',
+    ['w'] = 1280,
+    ['h'] = 720
+  },
+  ['HDTV 1080p'] = {
+    ['label'] = 'HDTV 1080p 1920x1080 (16:9)',
+    ['w'] = 1920,
+    ['h'] = 1080
+  },
+  ['Cinema TV'] = {
+    ['label'] = 'Cinema TV 2560x1080 (21:9)',
+    ['w'] = 2560,
+    ['h'] = 1080
+  },
+  ['2K'] = {
+    ['label'] = '2K 2048x1152 (16:9)',
+    ['w'] = 2048,
+    ['h'] = 1152
+  },
+  ['4K'] = {
+    ['label'] = '4K 4096x2304 (16:9)',
+    ['w'] = 4096,
+    ['h'] = 2304
+  }
+}
+
+local framerates = {'15', '16', '23.98', '24', '25', '29,97', '30', '48', '50', '59.94', '60'}
+
+local formats = {
+  ['AVI'] = {
+    ['extension'] = 'avi',
+    ['codecs'] = {'mpeg4', 'h263p', 'h264', 'mpeg2video', 'hevc', 'vp9'}
+  },
+  ['Matroska'] = {
+    ['extension'] = 'mkv',
+    ['codecs'] = {'h264', 'h263p', 'mpeg4', 'mpeg2video', 'hevc', 'vp9'}
+  },
+  ['WebM'] = {
+    ['extension'] = 'webm',
+    ['codecs'] = {'vp9', 'h263p', 'h264', 'mpeg4', 'mpeg2video', 'hevc'}
+  },
+  ['MP4'] = {
+    ['extension'] = 'mp4',
+    ['codecs'] = {'h264', 'h263p', 'mpeg4', 'mpeg2video'}
+  },
+  ['QuickTime'] = {
+    ['extension'] = 'mov',
+    ['codecs'] = {'h264', 'h263p', 'mpeg4', 'mpeg2video'}
+  }
+}
+
+res_list = {}
+for i, v in pairs(resolutions) do
+  table.insert(res_list, v['label'])
+end
+
+table.sort(res_list)
+
+format_list = {}
+for i,v in pairs(formats) do
+  table.insert(format_list, i)
+end
+
+table.sort(format_list)
+
+codec_list = formats['AVI']['codecs']
```

Comment:
Perhaps a comment explaining why you used this instead of the loop as above.  I figured it out, but a comment would make it easier for someone else

Reply:
done